Sources/Purchasing/StoreKit2/StoreKit2TransactionListener.swift

```diff
@@ -88,15 +96,16 @@ private extension StoreKit2TransactionListener {
             )
 
         case .verified(let verifiedTransaction):
-            await finish(transaction: verifiedTransaction)
+            let customerInfo = try await self.finish(transaction: verifiedTransaction)
 
-            return verifiedTransaction
+            return (verifiedTransaction, customerInfo)
         }
     }
 
-    func finish(transaction: StoreKit.Transaction) async {
+    func finish(transaction: StoreKit.Transaction) async throws -> CustomerInfo? {
         await transaction.finish()
-        delegate?.transactionsUpdated()
+
+        return try await delegate?.transactionsUpdated()
```

Comment:
`CustomerInfo` in the return here is nil only if the delegate is nil, right? Since the method in the delegate has a non-optional return value. 

Maybe we could make that more explicit with a `guard let delegate = self.delegate else { return nil }`